packages/core/src/subgraph/getConfirmDataFromGraph.ts

```diff
@@ -0,0 +1,62 @@
+import { ethers } from "ethers";
+import { GraphQLClient, gql } from 'graphql-request'
+import { config } from "../config.js";
+
+/**
+ * Fetches the confirmation data for multiple challenge assertions from the graph.
+ * 
+ * @param {number[]} assertionIds - An array of assertion IDs for which confirmation data is required.
+ * @returns {Promise<{string[], string}>} A promise that resolves to a tuple containing an array of `bytes32` confirmation data and a `bytes32` confirmation hash.
+ */
+export async function getConfirmDatafromGraph(assertionIds: number[]): Promise<{confirmData: string[], confirmHash: string}> {
+    
+    //initialize graph client
+    const client = new GraphQLClient(config.subgraphEndpoint);
+
+    //query for each assertion id (aka nodeNum)
+    let confirmDataArray: string[] = [];
+    for (let i = 0; i < assertionIds.length; i++) {
+        
+        //craft query
+        const query = gql`
+            query NodeConfirmationQuery {
+                nodeConfirmation(id: ${assertionIds[i]}) {
+                    id
+                    confirmData
+                }
+            }
```

Comment:
This should be done in a query for an array of nodeConfirmations like 

```
query MyQuery {
  nodeConfirmations(where: {id_in: ["1","2","3"]}) {
    sendRoot
    nodeNum
    confirmData
    id
    blockHash
  }
}
```

Reply:
Refactored the graphql query to make 1 call instead of multiple.